src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,189 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ * Checks that there is no whitespace before a colon token of a switch case or default.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NoWhitespaceBeforeCaseDefaultColon&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *   {
+ *     switch(1) {
+ *         case 1 : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *         case 2: // OK
+ *             break;
+ *         default : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *     }
+ *
+ *     switch(2) {
+ *         case 2: // OK
+ *             break;
+ *         default: // OK
+ *             break;
```

Comment:
I think it would be good to show a line wrapped example here, like you have done in your test input.

Reply:
Done.

Reply:
All we really need to do is:

```java
         case 2: // OK
             break;
         default
                      : //  violation
```
Also, I don't think we need two different examples for this check. We can probably just merge these two examples.

Reply:
Done

Reply:
@nmancus1 Is it resolved? Please comment on others too if they are resolved.

Reply:
Resolved.